File: lib/issue_tracker/models/mysql/issue.rb
require 'issue_tracker/models/mysql/model'

class Issue < Model

  property :project_id, Integer, max: 6, unsigned: true, required: true
  property :name, Varchar, max: 25, required: true
  property :description, Text
  property :parent, Integer, max: 6, unsigned: true
  property :type, Integer, max: 3, unsigned: true, required: true
  property :project, Integer, max: 3, unsigned: true, required: true
  property :state, Integer, max: 3, unsigned: true, required: true
  property :sprint, Integer, max: 6, unsigned: true
  property :user_assigned, Integer, max: 6, unsigned: true, required: true
  property :user_created, Integer, max: 6, unsigned: true, required: true

end

File: public/js/lib/issue.js


class Issue {
    
    constructor(data) {
        this.id = parseInt(data.id);
        this.parent = parseInt(data.parent);
        this.name = data.name;
        this.description = data.description === null ? '' : data.description;
        this.project = parseInt(data.project);
        this.project_id = parseInt(data.project_id);
        this.type = parseInt(data.type);
        this.user_created = parseInt(data.user_created);
        this.user_assigned = parseInt(data.user_assigned);
        this.state = parseInt(data.state);
        this.sprint = parseInt(data.sprint);
    }

    setState(state) {
        this.state = state;
    }
}

